src/MamoStakingStrategy.sol

```diff
@@ -0,0 +1,285 @@
+
+            // Swap reward tokens to MAMO
+            // Get tickSpacing from the pool
+            address poolAddress = rewardTokens[i].pool;
+            int24 tickSpacing = IPool(poolAddress).tickSpacing();
+
+            // Get quote from quoter to calculate minimum amount out with slippage
+            (uint256 amountOut,,,) = quoter.quoteExactInputSingle(
+                IQuoter.QuoteExactInputSingleParams({
+                    tokenIn: address(rewardToken),
+                    tokenOut: address(mamoToken),
+                    amountIn: rewardBalance,
+                    tickSpacing: tickSpacing,
+                    sqrtPriceLimitX96: 0
+                })
+            );
+
+            // Apply slippage tolerance to get minimum amount out
+            uint256 accountSlippage = getAccountSlippage();
+            uint256 amountOutMinimum = (amountOut * (10000 - accountSlippage)) / 10000;
+
+            // Approve DEX router to spend reward tokens
+            rewardToken.forceApprove(address(dexRouter), rewardBalance);
+
+            // Create swap parameters
+            ISwapRouter.ExactInputSingleParams memory swapParams = ISwapRouter.ExactInputSingleParams({
+                tokenIn: address(rewardToken),
+                tokenOut: address(mamoToken),
+                tickSpacing: tickSpacing,
+                recipient: address(this),
+                deadline: block.timestamp + 300,
+                amountIn: rewardBalance,
+                amountOutMinimum: amountOutMinimum,
+                sqrtPriceLimitX96: 0 // No price limit
+            });
+
+            // Execute swap and capture actual amount out
+            uint256 actualAmountOut = dexRouter.exactInputSingle(swapParams);
+
+            emit CompoundRewardTokenProcessed(address(rewardToken), rewardBalance, actualAmountOut);
+        }
+
+        // Stake all MAMO
+        uint256 totalMamo = mamoToken.balanceOf(address(this));
+        _stakeMamo(totalMamo);
+
+        emit Compounded(totalMamo);
+    }
+
+    /**
+     * @notice Reinvest rewards by staking MAMO and depositing other rewards to ERC20 strategies
+     * @param rewardStrategies Array of strategy addresses for each reward token (must match rewardTokens order)
+     * @dev Claims rewards and then reinvests them. We trust the backend to provide correct user-owned strategies.
+     */
+    function reinvest(address[] calldata rewardStrategies) external onlyBackend {
+        multiRewards.getReward();
+
+        MamoStakingRegistry.RewardToken[] memory rewardTokens = stakingRegistry.getRewardTokens();
+        require(rewardStrategies.length == rewardTokens.length, "Strategies length mismatch");
+
+        uint256 mamoBalance = mamoToken.balanceOf(address(this));
+
+        // Stake MAMO
+        _stakeMamo(mamoBalance);
+
+        // Process each reward token
+        for (uint256 i = 0; i < rewardTokens.length; i++) {
+            IERC20 rewardToken = IERC20(rewardTokens[i].token);
```

Comment:
if we are calling `_stakeMamo` on line 243, we can again continue here right if `rewardToken` == `mamoToken`

shouldn't this function fail since `stakeMamo`  will deposit all the `mamo` and then we would try to call deposit on the same strategy but we will 0 balance then

Reply:
nvm it wouldnt fail because of line 250, but i wonder if we should be explicit in continuing if token is mamo